inspectit-oce-agent/src/system-test/java/rocks/inspectit/oce/metrics/ClassLoaderMetricsSysTest.java

```diff
@@ -7,32 +7,36 @@
 import java.lang.management.ManagementFactory;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 public class ClassLoaderMetricsSysTest extends MetricsSysTestBase {
 
     private static final ViewManager viewManager = Stats.getViewManager();
 
     @Test
-    public void testClassLoaderMetricCapturing() throws Exception {
-        ViewData loadedData = viewManager.getView(View.Name.create("jvm/classes/loaded"));
-        ViewData unloadedData = viewManager.getView(View.Name.create("jvm/classes/unloaded"));
-        assertThat(loadedData.getAggregationMap()).isNotEmpty();
-        assertThat(unloadedData.getAggregationMap()).isNotEmpty();
-
-        Map.Entry<List<TagValue>, AggregationData> loaded = loadedData.getAggregationMap().entrySet().stream().findFirst().get();
-        Map.Entry<List<TagValue>, AggregationData> unloaded = unloadedData.getAggregationMap().entrySet().stream().findFirst().get();
-
-        //ensure that tags are present
-        assertThat(loaded.getKey()).isNotEmpty();
-        assertThat(unloaded.getKey()).isNotEmpty();
-
-        //ensure that the values are sane
-        long loadedVal = ((AggregationData.LastValueDataLong) loaded.getValue()).getLastValue();
-        long unloadedVal = ((AggregationData.LastValueDataLong) unloaded.getValue()).getLastValue();
-
-        assertThat(loadedVal).isLessThanOrEqualTo(ManagementFactory.getClassLoadingMXBean().getTotalLoadedClassCount());
-        assertThat(unloadedVal).isLessThanOrEqualTo(ManagementFactory.getClassLoadingMXBean().getUnloadedClassCount());
+    public void testClassLoaderMetricCapturing() {
+        await().atMost(5, TimeUnit.SECONDS).untilAsserted(() -> {
+            ViewData loadedData = viewManager.getView(View.Name.create("jvm/classes/loaded"));
```

Comment:
should you also call here  `TestUtils.waitForOpenCensusQueueToBeProcessed();` first? At least you are ensuring that queue is drained before you check? Maybe actually outside the await()!

Reply:
ahh ok this is fixed with:

```
@BeforeEach
    void flushMetrics() {
```